File: Exercícios sobre POO/Empresa 8/Empresa em python/empresa.py
class Empresa:
    def __init__(self, nome, cnpj) -> None:
        self.nome = nome
        self.cnpj = cnpj
        self.funcionarios = []

    def contratar_funcionarios(self, funcionario):
        self.funcionarios.append(funcionario)
    
    def listar_funcionarios(self):
        print(f"Funcionarios da empresa {self.nome}:")
        for funcionario in self.funcionarios:
            print(funcionario.nome)

File: Exercícios sobre POO/Empresa 8/Empresa em java/src/Empresa.java
import java.util.ArrayList;

public class Empresa {
    private String nome;
    private String cnpj;
    private ArrayList<String> funcionarios = new ArrayList<>();

    public Empresa(String nome, String cnpj){
        this.nome = nome;
        this.cnpj = cnpj;
    }

    public void contratar_funcionarios(Funcionario funcionario){
        this.funcionarios.add(funcionario.getNome());
    }

    public void listar_funcionarios(){
        System.out.println("Funcionarios da empresa " + this.nome + ":");
        for(String funcionario : funcionarios){
            System.out.println(funcionario);
        }
    }
}
